### PR TITLE
Integration test 

### LINKED ISSUE #18: Add integration tests
To start, run a few variations of the String load and confirm output, maybe including kgx validation?

### PR DESCRIPTION
**Pull request by kevinschaper**
This should run all of the StringDB examples that don't require the map (that could be added if we make a small example map data file to commit to the repository), and then run the kgx validator against them

Right now, this is marked as a draft because it's having a problem when two ingest runs are reading from the same file.

closes #18

### CODE REVIEW

File: examples/string/protein-links-detailed.py [hunk 1]
```diff
@@ -22,4 +24,4 @@
         relation='IAO:0002436'
     )
 
-    koza.write(source_name, [protein_a, protein_b, pairwise_gene_to_gene_interaction])
+    collect(source_name, protein_a, protein_b, pairwise_gene_to_gene_interaction)
```
Comment (kshefchek) [inline examples/string/protein-links-detailed.py:27]: we should probably go with write long term? more explicit than collect. I know i'm the one who originally called this collect - just noting here

File: tests/integration/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        ),
+    ],
+)
+def test_examples(ingest, output_names, output_format):
+
+    source = f"examples/{ingest}/metadata.yaml"
+    output_suffix = str(output_format).split('.')[1]
+    output_dir = f"./test-output/{ingest}-{output_suffix}"
+
+    output_files = [
```
Comment (kshefchek) [inline tests/integration/test_examples.py:68]: consider simplifying to:

    output_files = [
        output_file
        for file in output_names
        for output_file in [
            f"{output_dir}/{file}_nodes.{output_suffix}",
            f"{output_dir}/{file}_edges.{output_suffix}",
        ]
    ]
```

but either works!

**Comment by kshefchek** [comment]
this looks like an issue with importlib caching transform_code ('protein-links-detailed') and ignoring the new sys.path, so it ends up running the loop version in declarative mode which will loop infinitely.  taking a closer look at how to fix this

**Comment by kshefchek** [comment]
it looks like the issue here is we're treating the ingest files as modules and there are name clashes when running a list of sources with the same file name.  There are a couple options I can think of 

1. Treat the parent folder as a package
2. Document that two ingest files cannot have the same name and make sure for tests like these the app object and be set up and torn down between tests

I think 1 gets messy and vote for 2, @kevinschaper curious your thoughts

File: koza/app.py [hunk 3]
```diff
@@ -125,6 +125,9 @@ def process_sources(self):
             # close the writer when the source is done processing
             self.writer_registry[source_file.config.name].finalize()
 
+            # remove directory from sys.path to prevent name clashes
+            sys.path.remove(str(parent_path))
```
Comment (kshefchek) [inline koza/app.py:129]: this fixes 3 out of the 6 tests

**Comment by kevinschaper** [comment]
I'm good with staying away from option 1, for option 2, could we also keep track of any source names that have previously run if the app is alive through more than one ingest and throw an explicit error?  

(or maybe with that last fix, it's not an issue?)

**Comment by kshefchek** [comment]
> (or maybe with that last fix, it's not an issue?)

yes this looks to fix it.  I noticed the app is already being reset between runs, so I think it's the sys variable that is silently global

File: examples/string/protein-links-detailed.py (hunk 1 above)
Comment (kevinschaper) [reply examples/string/protein-links-detailed.py:27]: Sounds good to me, rather than doing it here, I made issue #21

**Comment by kevinschaper** [comment]
I'm not sure what's going on with the tests, the kgx import in the tests is failing on link_model?

**Comment by sierra-moxon** [comment]
@kevinschaper - linkml updated last week and I'm working on getting KGX migrated to the new version.  I'm also trying to decide if I should pin linkml version in KGX and BMT to avoid unexpected failures like this in the future...any feedback on that idea welcome! :)

**Comment by kevinschaper** [comment]
@sierra-moxon bmt has the option to use older versions, right? so that's probably an easier choice to let it just use the latest. I'm likely to be happier with kgx also just using the latest, but maybe that would cause problems for others if there's no option to specify a version?

**Comment by kevinschaper** [comment]
Finally passing!

**Pull request merged**